sotawhat/sotawhat.py

```diff
@@ -129,7 +141,7 @@ def has_number(sent):
             return True
         try:
             value = int(token)
-        except:
+        except Exception:
```

Comment:
I'm curious about what purpose this change serves? It seems like maybe catching `ValueError` and possibly even `TypeError` would be appropriate but going from (essentially) `BaseException` to `Exception` only seems to skip handling three exceptions that seem unlikely to be raised here?

* https://docs.python.org/3/library/exceptions.html#exception-hierarchy

```
BaseException
 +-- SystemExit
 +-- KeyboardInterrupt
 +-- GeneratorExit
 +-- Exception
      +-- StopIteration
      +-- StopAsyncIteration
      +-- ArithmeticError
      |    +-- FloatingPointError
      |    +-- OverflowError
      |    +-- ZeroDivisionError
      +-- AssertionError
      +-- AttributeError
      +-- BufferError
      +-- EOFError
      +-- ImportError
      ...
```

```python
>>> int(lambda: None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: int() argument must be a string, a bytes-like object or a number, not 'function'

>>> int('not a number')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: 'not a number'

```

Reply:
PEP8 discourages usage of bare exceptions - https://www.python.org/dev/peps/pep-0008/

Its precisely the KeyboardInterrupt exception that I want to allow the script to handle. The changes in this PR allow for calling sotawhat from the terminal using `sotawhat ...` globally in any directory. However, if for some reason you need to escape before complete execution, you would normally use (CTRL + C / CMD + C) to force the KeyboardInterrupt, which terminates the program.

Using a bare exception, even such interrupts are ignored.

Reply:
Using `except Exception` is essentially a bare exception though. Your reasoning seems odd, why would a user need to use a `KeyboardInterrupt` inside of the `has_number` function exactly?

Reply:
It seems very unlikely that a user would hit CTRL+C at the exact moment that this single line of code is run:

```python
value = int(token)
```

The two most likely exceptions were mentioned in my original comment, why not just use them? 

> ***When catching exceptions, mention specific exceptions whenever possible instead of using a bare except: clause.***

* http://pep8.org/#programming-recommendations

Reply:
As I mentioned, I conformed to the PEP8 guidelines. 

As to your concerns of catching the other two exceptions, they are invalid. I do not know why any user would pass a lambda expression or a specific nan string when specifically asked for an integer input. 

In case this was done on purpose, argparse would raise an error first before executing the main function, as it also casts the input value to type int.

Reply:
What exceptions do you think could be raised by this one line of code? 

```python
value = int(token)
```

Reply:
Does it matter which exception is raised? No. The original code simply skips that row. Therefore I did the same.

I chose to catch Exception instead of using a bare except statement, in order to follow PEP8 guidelines. 

Why is this such an issue?

Reply:
> ***When catching exceptions, mention specific exceptions whenever possible instead of using a bare except: clause.***

* http://pep8.org/#programming-recommendations

If you were actually following PEP8, you would specify the exact exceptions that would be raised by that line of code. The two exceptions that are raised are `ValueError` and `TypeError` but you seem to want to catch all of these as well without any reason?

```
 +-- Exception
      +-- StopIteration
      +-- StopAsyncIteration
      +-- ArithmeticError
      |    +-- FloatingPointError
      |    +-- OverflowError
      |    +-- ZeroDivisionError
      +-- AssertionError
      +-- AttributeError
      +-- BufferError
      +-- EOFError
      +-- ImportError
      |    +-- ModuleNotFoundError
      +-- LookupError
      |    +-- IndexError
      |    +-- KeyError
      +-- MemoryError
      +-- NameError
      |    +-- UnboundLocalError
      +-- OSError
      |    +-- BlockingIOError
      |    +-- ChildProcessError
      |    +-- ConnectionError
      |    |    +-- BrokenPipeError
      |    |    +-- ConnectionAbortedError
      |    |    +-- ConnectionRefusedError
      |    |    +-- ConnectionResetError
      |    +-- FileExistsError
      |    +-- FileNotFoundError
      |    +-- InterruptedError
      |    +-- IsADirectoryError
      |    +-- NotADirectoryError
      |    +-- PermissionError
      |    +-- ProcessLookupError
      |    +-- TimeoutError
      +-- ReferenceError
      +-- RuntimeError
      |    +-- NotImplementedError
      |    +-- RecursionError
      +-- SyntaxError
      |    +-- IndentationError
      |         +-- TabError
      +-- SystemError
      |    +-- UnicodeError
      |         +-- UnicodeDecodeError
      |         +-- UnicodeEncodeError
      |         +-- UnicodeTranslateError
      +-- Warning
           +-- DeprecationWarning
           +-- PendingDeprecationWarning
           +-- RuntimeWarning
           +-- SyntaxWarning
           +-- UserWarning
           +-- FutureWarning
           +-- ImportWarning
           +-- UnicodeWarning
           +-- BytesWarning
           +-- ResourceWarning
```

Reply:
And by catching those two exceptions, I would simply add a few more continues. What benefit would catching the specific exceptions be ?

It would be an addition of checks with no additional benefit.

Reply:
A few more continues? Where did you get that idea? Have you never caught multiple exceptions before?

```python
try:
    value = int(token)
except (ValueError, TypeError):
    continue

```

Reply:
Generally, I handle different exceptions on different lines. For this case, the above works just as well.